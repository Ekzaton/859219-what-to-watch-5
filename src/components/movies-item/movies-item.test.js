import React from "react";
import renderer from "react-test-renderer";
import {Route, BrowserRouter} from "react-router-dom";

import mockPromo from "../../mocks/promo";

import MoviesItem from "./movies-item";

describe(`MoviesItem snapshot testing`, () => {

  const movieItemComponent = renderer.create(
      <BrowserRouter>
        <Route>
          <MoviesItem
            movie={mockPromo}
            onMovieEnter={() => {}}
            onMovieLeave={() => {}}
            isPlaying={false}
          />
        </Route>
      </BrowserRouter>
  );

  it(`renders component correctly`, () => {
    expect(movieItemComponent.toJSON()).toMatchSnapshot();
  });
});
