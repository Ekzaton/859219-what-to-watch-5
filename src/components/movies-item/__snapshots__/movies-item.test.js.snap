// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesItem snapshot testing renders component correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    className="small-movie-card__image"
    href="/films/1"
    onClick={[Function]}
  >
    <video
      height="175"
      muted={true}
      poster="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
      src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
      width="280"
    />
  </a>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Bronson
    </a>
  </h3>
</article>
`;
